### PR TITLE
[Bug](planner) fix core dump when lateral view above union node and have predicate

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

fix core dump when lateral view above union node and have predicate

```cpp
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /mnt/disk1/compile/doris-1.2-lts/be/src/common/signal_handler.h:420
 1# os::Linux::chained_handler(int, siginfo_t*, void*) in /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-0.1.al8.x86_64/lib/server/libjvm.so
 2# JVM_handle_linux_signal in /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-0.1.al8.x86_64/lib/server/libjvm.so
 3# signalHandler(int, siginfo_t*, void*) in /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-0.1.al8.x86_64/lib/server/libjvm.so
 4# 0x00007FD9104ACC40 in /lib64/libc.so.6
 5# doris::vectorized::DataTypeNullable::equals(doris::vectorized::IDataType const&) const at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/data_types/data_type_nullable.cpp
:147
 6# void doris::vectorized::MutableBlock::merge<doris::vectorized::Block>(doris::vectorized::Block&&) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/core/block.h:471
 7# doris::vectorized::VUnionNode::get_next_materialized(doris::RuntimeState*, doris::vectorized::Block*) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/exec/vunion_node.c
pp:156
 8# doris::vectorized::VUnionNode::get_next(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/exec/vunion_node.cpp:241
 9# doris::ExecNode::get_next_after_projects(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/exec/exec_node.cpp:852
10# doris::vectorized::VTableFunctionNode::get_expanded_block(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/exec/vtable_function_node.cpp:117
11# doris::vectorized::VTableFunctionNode::get_next(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/exec/vtable_function_node.cpp:86
12# doris::ExecNode::get_next_after_projects(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/exec/exec_node.cpp:852
13# doris::vectorized::AggregationNode::get_next(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/vec/exec/vaggregation_node.cpp:500
14# doris::ExecNode::get_next_after_projects(doris::RuntimeState*, doris::vectorized::Block*, bool*) at /mnt/disk1/compile/doris-1.2-lts/be/src/exec/exec_node.cpp:852
15# doris::PlanFragmentExecutor::get_vectorized_internal(doris::vectorized::Block**) at /mnt/disk1/compile/doris-1.2-lts/be/src/runtime/plan_fragment_executor.cpp:358
16# doris::PlanFragmentExecutor::open_vectorized_internal() at /mnt/disk1/compile/doris-1.2-lts/be/src/runtime/plan_fragment_executor.cpp:306
17# doris::PlanFragmentExecutor::open() at /mnt/disk1/compile/doris-1.2-lts/be/src/runtime/plan_fragment_executor.cpp:259
18# doris::FragmentExecState::execute() at /mnt/disk1/compile/doris-1.2-lts/be/src/runtime/fragment_mgr.cpp:253
19# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::PlanFragmentExecutor*)>) at /mnt/disk1/compile/doris-1.2-lts/be/src/runtime/fragment_mgr.cpp:500
20# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::PlanFragmentExecutor*)>)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at /opt/ldb_toolchain/include/c++/11/bits/std_function.h:291
21# doris::ThreadPool::dispatch_thread() at /mnt/disk1/compile/doris-1.2-lts/be/src/util/threadpool.cpp:543
22# doris::Thread::supervise_thread(void*) at /mnt/disk1/compile/doris-1.2-lts/be/src/util/thread.cpp:455
23# start_thread in /lib64/libpthread.so.0
24# clone in /lib64/libc.so.6
```

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run arm

**Comment by BiteTheDDDDt** [comment]
run arm

File: regression-test/suites/query_p0/lateral_view/test_with_view.groovy [hunk 1]
```diff
@@ -32,4 +32,29 @@ suite("lateral_view_with_inline_view") {
...
+                    select
+                        3 hour,
+                        10 money
+                ) t lateral view explode([23,24]) tmp1 as e1
+            where
+                hour <= e1
+        ) tt
+    group by
+        1;
+    """
```
Comment (yiguolei) [inline regression-test/suites/query_p0/lateral_view/test_with_view.groovy:59]: add order by to make the test stable

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

File: be/src/vec/core/block.cpp [hunk 2]
```diff
@@ -400,12 +400,25 @@ size_t Block::allocated_bytes() const {
 std::string Block::dump_names() const {
     std::stringstream out;
     for (auto it = data.begin(); it != data.end(); ++it) {
-        if (it != data.begin()) out << ", ";
```
Comment (HappenLee) [inline be/src/vec/core/block.cpp:403]: change stringstream to fmt::memory_buffer to improve the performance

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.84 seconds
 stream load tsv:          451 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         22 seconds loaded 2358488459 Bytes, about 102 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230322010604_clickbench_pr_118241.html

**Comment by BiteTheDDDDt** [comment]
run arm

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**